1788 admin unix socket

Firstyear opened the pull request:
Fixes #1788 - This adds an administrative unix socket to the main kanidmd daemon allowing tasks to be performed while the server is online. Currently this is limited to the recover-account workflow. The socket requires the calling user to be the same uid as the kanidmd process or root. 

Checklist

- [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ x ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ x ] book chapter included (if relevant)
- [ ] design document included (if relevant)

Firstyear commented:
@yaleman Should we add default config path to profilse to make the commands easier to use here?

yaleman commented on tools/orca/src/ipa.rs:
```diff
@@ -1,5 +1,5 @@
+use hashbrown::{HashMap, HashSet};
```
Isn't this just a cheeky workaround for the `std::collections::HashMap` ban thing? The `hashbrown` crate says it's identical?

> Since Rust 1.36, this is now the HashMap implementation for the Rust standard library.

Firstyear replied:
It uses a different hasher though in std lib. hashbrown uses ahash, but std uses siphash which is significantly slower. That's why hashbrown is preferred.

yaleman commented on unix_integration/src/daemon.rs:
```diff
@@ -729,12 +729,12 @@ async fn main() -> ExitCode {
                                             // all good!
                                         } else {
```
Shouldn't this just be `if ucred.uid() != 0 { warn... continue}`

yaleman commented:
> @yaleman Should we add default config path to profilse to make the commands easier to use here?

Yeah, I think there should be a default with the option to modify it, say:

- dev `../../examples/insecure_server.toml` (or `<the package root>/examples/insecure_server.toml` if we can pick that up)
- non-container `/etc/kanidm/kanidmd.toml`
- container `/data/kanidmd.toml`

Firstyear commented:
> > @yaleman Should we add default config path to profilse to make the commands easier to use here?
> 
> Yeah, I think there should be a default with the option to modify it, say:
> 
> * dev `../../examples/insecure_server.toml` (or `<the package root>/examples/insecure_server.toml` if we can pick that up)
> * non-container `/etc/kanidm/kanidmd.toml`
> * container `/data/kanidmd.toml`

Done, added it to profiles, -c always overrides, and we fall back to the default. Added the defaults in line with what the book suggests for config paths.
